src/main/kotlin/com/openlattice/data/storage/PostgresEntityDataQueryService.kt

```diff
@@ -557,43 +556,56 @@ class PostgresEntityDataQueryService(
-        val numUpdates = hds.connection.use { connection ->
-            // Delete properties from S3
-            authorizedPropertyTypes.map { property ->
-                if (property.value.datatype == EdmPrimitiveTypeKind.Binary) {
-                    deletePropertyOfEntityFromS3(entitySetId, entityKeyIds, property.key)
-                }
+        // Delete properties from S3
+        authorizedPropertyTypes.map { property ->
+            if (property.value.datatype == EdmPrimitiveTypeKind.Binary) {
+                deletePropertyOfEntityFromS3(entitySetId, entityKeyIds, property.key)
             }
+        }
 
+        val partitions = partitionsInfo.partitions.toList()
+        val partitionVersion = partitionsInfo.partitionsVersion
+        val numUpdates = entityKeyIds
+                .groupBy { getPartition(it, partitions) }
+                .map { (partition, entities) ->
+                    deleteEntities(entitySetId, entities, authorizedPropertyTypes, partition, partitionVersion)
+                }.sum()
+
+        return WriteEvent(System.currentTimeMillis(), numUpdates)
+    }
+
+    private fun deleteEntities(
+            entitySetId: UUID,
+            entities: Collection<UUID>,
+            authorizedPropertyTypes: Map<UUID, PropertyType>,
+            partition: Int,
+            partitionVersion: Int
+    ): Int {
+        return hds.connection.use { connection ->
             connection.autoCommit = false
-            val partitions = partitionsInfo.partitions.toList()
+
             val propertyTypesArr = PostgresArrays.createUuidArray(connection, authorizedPropertyTypes.keys)
-            entityKeyIds
-                    .groupBy { getPartition(it, partitions) }
-                    .map { (partition, entities) ->
-                        val idsArr = PostgresArrays.createUuidArray(connection, entities)
-
-                        // Acquire entity key id locks
-                        val rowLocks = connection.prepareStatement(lockEntitiesSql)
-                        rowLocks.setArray(1, idsArr)
-                        rowLocks.setInt(2, partition)
-                        rowLocks.executeQuery()
-
-                        // Delete entity properties from data table
-                        val ps = connection.prepareStatement(deletePropertiesOfEntitiesInEntitySet)
-                        ps.setObject(1, entitySetId)
-                        ps.setArray(2, idsArr)
-                        ps.setInt(3, partition)
-                        ps.setInt(4, partitionsInfo.partitionsVersion)
-                        ps.setArray(5, propertyTypesArr)
-
-                        val count = ps.executeUpdate()
-                        connection.commit()
-                        count
-                    }.sum()
-        }
 
-        return WriteEvent(System.currentTimeMillis(), numUpdates)
+            val idsArr = PostgresArrays.createUuidArray(connection, entities)
+
+            // Acquire entity key id locks
```

Comment:
Just curious about postgres things here: we don't want to turn this into a batch call because we need to lock all these rows right?

Reply:
I think, we also had the same problem , if it were a batch call, because of the connection timeout possibility